artifactory/services/upload.go

```diff
@@ -159,19 +244,22 @@ func collectFilesForUpload(uploadParams UploadParams, producer parallel.Runner,
 			buildProps += vcsProps
 		}
 		uploadData := UploadData{Artifact: artifact, TargetProps: props, BuildProps: buildProps}
-		task := artifactHandlerFunc(uploadData)
 		if progressMgr != nil {
-			progressMgr.IncGeneralProgressTotalBy(1)
+			if uploadParams.Archive != "" {
+				progressMgr.IncGeneralProgressTotalBy(2)
+			} else {
+				progressMgr.IncGeneralProgressTotalBy(1)
+			}
 		}
```

Comment:
Since the above code is duplicated twice in this file, let's move it into a provate function and resuse it.
Let's also explain why ```IncGeneralProgressTotalBy``` receives 2 in the case of an archive.